### PR TITLE
fix test suite

### PR DESCRIPTION
**Pull request by charlierudolph**
@jbpros @samccone 

All the undefined steps when running cucumber-js and cucumber are very annoying to work with.

Added the strict flag and got everything passing.

* moved cucumber-tck to root level so when running cucumber-js, they are not run
* moved features written in ruby to ruby_features
* introduced a new ruby cucumber profile for running the compliance tests
* fixed steps that were undefined in both

### CODE REVIEW

**Comment by charlierudolph** [comment]
Also moved legacy features into their own folder. Are these even worth keeping around? If so should they be added to the test suite?

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ The official way of running them is through Cucumber-ruby and Aruba. Ruby and Bu
 
     $ git submodule update --init
     $ bundle
-    $ rm -rf doc; ARUBA_REPORT_DIR=doc cucumber features/cucumber-tck -r features
+    $ rm -rf doc; ARUBA_REPORT_DIR=doc cucumber -p compliance
```
Comment (charlierudolph) [inline CONTRIBUTE.md]: Specifying the ARUBA_REPORT_DIR is causing this to fail on my machine.
Had to add bcat and rdiscount to the gemfile just to get it to run.
Then every test is hitting
```
undefined method `line' for  #<Cucumber::...>
```
Where it says the error is keeps changes with each test

File: features/with_bom.feature [hunk 2]
```diff
@@ -1,53 +0,0 @@
...
-    And I should not see any username
-
-
-  Scenario: User sees 'connecting' while connecting
-    Given I have had lagging api site
-    When I visit initial page
-    Then I should see connecting message
-    And I should not see any username
-
-
```
Comment (charlierudolph) [inline features/with_bom.feature]: This test had all undefined steps when run under either context. Any idea what this is for?

File: features/step_definition_snippets.feature [hunk 3]
```diff
@@ -1,35 +1,9 @@
...
-    Then a "Given" step definition snippet for /^I have (\d+) "([^"]*)" cucumbers$/ with 2 parameters is suggested
-
-  Scenario: multiple matching groups
-    Given a scenario with:
-      """
-      Given I have some "hekiri", "wild" and "regular" cucumbers
-      """
-    When Cucumber executes the scenario
-    Then a "Given" step definition snippet for /^I have some "([^"]*)", "([^"]*)" and "([^"]*)" cucumbers$/ with 3 parameters is suggested
-
```
Comment (charlierudolph) [inline features/step_definition_snippets.feature]: These were moved to a ruby feature.

File: CONTRIBUTE.md (hunk 1 above)
Comment (charlierudolph) [inline CONTRIBUTE.md]: This can be dealt with separately but wanted to bring it up

**Comment by charlierudolph** [comment]
@jbpros any thoughts? This would make things much nicer to work with.

**Comment by jbpros** [comment]
Thanks Charlie. There's definitely room for improvement in the organisation of the tests.

I like the ruby profiles, great idea. I also like the NPM scripts you added, makes things clearer. And obviously, fixing the undefined steps is the way to go!

Some comments:

1. cucumber-tck is supposed to be run by cucumber.js itself as well. Not all steps are implemented, but a good deal of them are. try `cucumber.js -r features -r cucumber-tck cucumber-tck` in your branch to see that. BTW, some steps which have moved are broken (wrong require paths to rpn_calculator).

2. Running `cucumber.js -r features/ -r cucumber-tck ruby_features/` also shows that `ruby_features` contains a few scenarios that have JS step definitions. I don't want to stop running those.

3. Not sure about moving cucumber-tck to the root folder. I personally prefer to keep things in one place; I'd rather see a new features/XXXX folder containing the specific feature files of Cucumber.js.

**Comment by charlierudolph** [comment]
1. Good to know. I'd personally prefer not to run it in javascript until its all green. I'm happy to work on finishing the conversion in another PR. 

2. I'd like to just see `ruby_features` removed. I'd suggest we just leave them separate for now and can slowly convert them over.

3. Okay until we get everything converted over to cucumber-js should we just have 
  * `features/cucumber-tck` (run with cucumber-ruby for now, to be run with cucumber-js in near future)
  * `features/javascript` (everything is here will be moved to the top level once all steps are implemented in javascript)
  * `features/ruby` (to be removed, once all steps defined)

In support of this I would love to get #303 in so the default cucumber-js profile runs just the javascript features. Can you please review or suggest who I should be asking for reviews?

**Comment by charlierudolph** [comment]
I'm just going to close this and concentrate on removing ruby altogether.

**Pull request closed without merging**